pytoda/datasets/_cache_datasource.py

```diff
@@ -0,0 +1,21 @@
+"""_CacheDatasource abstract implementation."""
+import tempfile
+import shutil
+import diskcache as dc
+
+
+class _CacheDatasource:
+    """
+    Supporting Datasets with underlying disk cache.
+
+    Suggested when handling datasets that can't fit in the device memory.
+    """
+
+    def __init__(self, size_limit=1073741824) -> None:
+        """Initialize a dataset using a disk cache with maximal size."""
+        self.cache_filepath = tempfile.mkdtemp()
+        self.cache = dc.Cache(self.cache_filepath, size_limit=size_limit)
```

Comment:
I would store the size_limit, we might use it on the object.
```python
self.size_limit = size_limit
```

Reply:
This is technically not needed as it's available through `CacheDatasource.cache.size_limit`, but I added it for visibility.